### ISSUE TITLE
Emergency Repair Generals Power is very rarely used

### ISSUE THREAD

**Issue by xezon** [TheSuperHackers/GeneralsGamePatch#707]
Emergency Repair Generals Power is very rarely used across all 3 factions. It repairs vehicles in a selected area. Check if there can be anything done to make it more attractive to use.

## Stats
```
SpecialPower SuperweaponEmergencyRepair
  ReloadTime = 240000

Object RepairVehiclesInArea_InvisibleMarker_Level1
  HealingAmount = 100
  Radius = 100
  KindOf = VEHICLE

Object RepairVehiclesInArea_InvisibleMarker_Level2
  HealingAmount = 200
  Radius = 100
  KindOf = VEHICLE

Object RepairVehiclesInArea_InvisibleMarker_Level3
  HealingAmount = 300
  Radius = 100
  KindOf = VEHICLE
```

### Proposal 1
Decrease reload time.

### Proposal 2
Increase healing amount.

### Proposal 3
Increase radius.

### Proposal 4
Repair more kinds of things, such as STRUCTURE or INFANTRY.

### Proposal 5
Increase the radius. Make the repair effect linger for 10 seconds (with visible decal). Repair the units over time instead of in one burst. Upgrading the ability increases how long the effect lasts (20, 30 seconds?).
Example implementation: https://github.com/commy2/zerohour/commit/ea284287498fd7733bcaeefca2f099713bb65e4d

**Comment by commy2** [comment]
Proposal 5

Increase the radius. Make the repair effect linger for 10 seconds (with visible decal). Repair the units over time instead of in one burst. Upgrading the ability increases how long the effect lasts (20,30 seconds?).

Example implementation is here: https://github.com/commy2/zerohour/commit/ea284287498fd7733bcaeefca2f099713bb65e4d

**Comment by ZekeDlyoung** [comment]
Maybe make it so that it heals/repairs everything, not just vehicles, so it can be used in more situations and not just tank spams.

**Comment by xezon** [comment]
One problem I have with this ability is the lack of immediate usefulness after cooldown. Unlike other abilities such as GPS Scrambler, Artillery Barrage, Spy Drone, et al, it may not be suitable to use straight away after power is ready to use. There may be no important battles going on yet. This will further increase its usage frequency and therefore decrease its impact on the match.

The same applies to the China Frenzy power.

**Comment by MTKing4** [comment]
What if it worked on full hp units as well? Grant it as an upgrade, and will have it repaired whenever it gets damaged later on, but also repair over time and not in one burst, this way there's always a use for it, just use it on an important unit right before going to battle

**Comment by xezon** [comment]
That sounds like very complicated logic if there is no native functionality to do that.

**Comment by MTKing4** [comment]
Yeah I don't know if it's possible, but maybe someone can find out

**Comment by xezon** [comment]
If it sounds difficult it is probably better to explore simpler solutions first.

**Comment by commy2** [comment]
What MTKing4 refers to is called "overrepair" in Company of Heroes.  As far as I can tell, it is not technically feasible without engine changes.

**Comment by MTKing4** [comment]
3 levels of (frenzy/Emergency Repair) is really pain in the ass to fix, I can't imagine how anyone would spend more than 1 point on this over any other gen point

In fact the only time when you spend 1 on it is when you have an extra point and don't know what to do with it.

The only way this changes is by making it so good that you take it instead of any other useful 3 level powe like artillery or A10 Or Cash Bounty , keep in mind, this WILL change the meta by a lot if achieved.

imagine having a build just to get (frenzy/emergency Repair) and leveled up, it has to be game changing to be attractive for players

**Possible Solution for Emergency Repair**

Make Cooldown 1 minute
Or even less than that

And/Or use commy's suggestion, have it repair over time rather than giving it all at once

**Comment by xezon** [comment]
I suspect if General Powers are reasonably useful like competing ones, then players will find situations to make use of them.

**Comment by Jundiyy** [comment]
Correct, we don't need to make them so good that they are the only GPS picked. If we make them average, players will pick them in more situations.
An easy example, is using Tox in an FFA. Players will save the last point to see if they need might need a Scud Launcher.

**Comment by ExiLe-ZH** [comment]
> I suspect if General Powers are reasonably useful like competing ones, then players will find situations to make use of them.

'Reasonably useful' would mean REALLY COMPETITIVE though, with only 7 points to spent till rank 5, dedicating 2 or even 3 points is A LOT.

**Comment by MTKing4** [comment]
> Correct, we don't need to make them so good that they are the only GPS picked. If we make them average, players will pick them in more situations.

They won't be the only ones picked, they just need to be super good to be equally as useful as artillery for example, so that people actually have a choice wether to go for artillery(or whatever) or Emergency Repair. otherwise, being average means nobody will pick them still, or at least not higher than level 1

**Comment by commy2** [comment]
ER (as well as Frenzy) works on allied players, so there could be some utility in team games.

**Comment by ExiLe-ZH** [comment]
> > Correct, we don't need to make them so good that they are the only GPS picked. If we make them average, players will pick them in more situations.
> 
> They won't be the only ones picked, they just need to be super good to be equally as useful as artillery for example, so that people actually have a choice wether to go for artillery(or whatever) or Emergency Repair. otherwise, being average means nobody will pick them still, or at least not higher than level 1

This is the reason why I keep thinking whether we should actually buff them massively (level 2 and 3), I know the idea might turn off some of you here (including myself initially), but if you think realistically, how can these powers ever make skipping Artillery or Carpet an easy choice?

**Comment by Jundiyy** [comment]
By average, it means that they would be picked, perhaps the words don't describe what we mean. I mean, just not op stuff. Like Frenzy is, 30% for 10s would be quite comparable to a level 1 Arty imo. Maybe even more.
Over here, perhaps doubling the amount it heals + lower reload.

**Comment by MTKing4** [comment]
> > > Correct, we don't need to make them so good that they are the only GPS picked. If we make them average, players will pick them in more situations.
> > 
> > 
> > They won't be the only ones picked, they just need to be super good to be equally as useful as artillery for example, so that people actually have a choice wether to go for artillery(or whatever) or Emergency Repair. otherwise, being average means nobody will pick them still, or at least not higher than level 1
> 
> This is the reason why I keep thinking whether we should actually buff them massively (level 2 and 3), I know the idea might turn off some of you here (including myself initially), but if you think realistically, how can these powers ever make skipping Artillery or Carpet an easy choice?

Totally agree, they need to be massively buffed

> just not op stuff

We won't let them be OP of course, but they need to be buffed big time to be equals

**Comment by xezon** [comment]
It is really hard to decide on this one because they are so many different tweaks possible. I do like Proposal 5 for its practicality and Proposal 4 for its versatility.

**Comment by ExiLe-ZH** [comment]
> By average, it means that they would be picked, perhaps the words don't describe what we mean. I mean, just not op stuff. Like Frenzy is, 30% for 10s would be quite comparable to a lebel 1 Arty imo. Maybe even more.

Basically this in terms of strength:
level 1 emergency/frenzy/cashhack = level 1 barrage/a10
level 2 emergency/frenzy/cashhack = level 2 barrage/a10
level 3 emergency/frenzy/cashhack = level 3 barrage/a10

I think currently level 1 frenzy and cashhack are good enough. Emergency level 1 sucks.

**Comment by Jundiyy** [comment]
I think increase the amount + lower the reload.

I have 2 opinions here.
1. Keep as is but reload is down to 2 min 30 or 2 mins.
2. Reload is 3 mins but heal is 150/300/450.

The problem with point 2 is that, players might not get level 3 as most units will have healed to maximum or close to with just level 2. Level 3 would only be needed for Ovies or Composite Armour units.
Whereas with point 1, even level 3 would be used since its always reloading quickly. Literally every battle since there will be many small hit and runs.

1 point to remember though. Emergency Repair is bad for China. Especially Inf, since it won't repair anything that is being repaired by Enthusiasm. Thyme fix.

**Comment by commy2** [comment]
Prop 5 could be modified to also repair structures (I think).

**Comment by xezon** [comment]
I tested Proposal 4 + 5 yesterday. Technically it looks robust, other than suffering from general repair issue TheSuperHackers/GeneralsGameCode#240 too.

Ability to repair structures is very nice, because it gives more utility and ultimately adds more strategic depth to the game. So for example it could try to help repair a damaged Internet Center or Strategy Center. With Proposal 5 the Structure repair is not instant, so the impact is not as sudden as if classic Emergency Repair was utilized on structures.

I think stacked repair is preferred with Proposal 5, because it is bad if the ability is used but has no effect on blob of units that already have another Repair source. If it stacks, then it effectively doubles Ambulance repair, roughly doubles upgraded Propaganda and more than triples Junk Repair.

Regardless of above fixes and improvements, I think Reload Time needs to be reduced (also for Frenzy ability). I have already mentioned, Emergency Repair and Frenzy suffer from immediate usefulness. It is unreasonable to expect that it is useful to deploy as soon as it is ready, contrary to attack Powers that require no combination of forces. Therefore I think a reduction from 4 minutes to 3 minutes could be appropriate. This gives a bit headroom to find a good time for deployment.

Another consideration could be to increase healing rates per level.

Structure repair rate should probably be close to Dozer repair rate.

**Comment by MTKing4** [comment]
I don't really like buildings repair too much, it feels odd and unnecessary, instead i think a 2 minute cool down would be better

**Comment by xezon** [comment]
> I don't really like buildings repair too much, it feels odd and unnecessary

Could you elaborate why it feels odd and unnecessary?

From my POV logically it is less odd than the USA Ambulance. Because the Ambulance repairs Vehicles, but real Ambulances do not do that. They just treat people on the way to the Hospital.

I tested Proposal 5 in the game and it was a natural experience to see repair taking affect on buildings. The repair terminology thematically also fits, so there should be no confusion about it. What could be odd, if Repair would also heal infantry, like the Ambulance does.

The goal is to make the Emergency Repair ability more useful, because it is rarely used now. So giving it more utility should help with that. I do not understand why you say it is unnecessary.

**Comment by commy2** [comment]
Reduced it to 3 minutes and buffed building repair further.

**Comment by commy2** [comment]
I had a moment.

https://github.com/TheSuperHackers/GeneralsGamePatch/assets/6576312/369c10c8-0597-4a7b-8a78-534e46c6e2de

Trick is to use
```ini
  Behavior = FireWeaponUpdate ModuleTag_22
    Weapon = EmergencyRepairWeapon
  End
```
and
```ini
Weapon EmergencyRepairWeapon
  PrimaryDamage = 5.0
  PrimaryDamageRadius = 150.0
  DamageDealtAtSelfPosition = Yes
  AttackRange = 100.0
  DamageType = HEALING
  DeathType = NORMAL
  WeaponSpeed = 600                     ;  dist/sec
  RadiusDamageAffects = ALLIES NOT_AIRBORNE
  DelayBetweenShots = 250                ; time between shots, msec
End
```
BUT.

But this will no longer make us able to heal extra for buildings.

**Comment by xezon** [comment]
And it will probably also heal Infantry.

**Comment by commy2** [comment]
Yeah, it will. Not good.

**Comment by MTKing4** [comment]
> Could you elaborate why it feels odd and unnecessary?

To me it looks like a feature redesign, too much change, which we were trying to avoid as much as possible, i consider it unnecessary because you can buff the repair feature without having to implement new features to it.

Adding new features has a lot of consequences that i fear have not been taken into consideration here, things like this will shift the meta by quite a bit, now you will have situations where people would use this to save his last building on a base race for example and win an otherwise lost game, or save a building long enough to build a (MBT, Quad, Avenger,) to save the day from a probably lost match as well (to kill a dragon, helix, or combat chinook), this is why i think it's unnecessary, it will introduce new mechanics that are not known to be desired.

**Comment by AIR-Z** [comment]
repair could grant automatic healing to a unit within a minute and + 50% instant heal
